src/Pact/Analyze/Check.hs

```diff
@@ -639,41 +686,50 @@ runExpParserOver exps parser = sequence $ exps <&> \meta -> case parser meta of
   Left err   -> Left (meta, err)
   Right good -> Right (Located (getInfo meta) good)
 
+typecheck :: Ref -> IO (Either CheckFailure (TopLevel Node))
+typecheck ref = do
+  (toplevel, tcState) <- runTC 0 False $ typecheckTopLevel ref
+  let failures = tcState ^. tcFailures
+      info = toplevelInfo toplevel
+  pure $ if Set.null failures
+            then Right toplevel
+            else Left $ CheckFailure info $ TypecheckFailure failures
+
 verifyFunctionProps
   :: ModuleName
   -> [Table]
+  -> [Capability]
   -> Ref
   -> Text
   -> [Located Check]
   -> IO [CheckResult]
-verifyFunctionProps modName tables ref funName props = do
-  (fun, tcState) <- runTC 0 False $ typecheckTopLevel ref
-  let failures = tcState ^. tcFailures
-
-  case fun of
-    TopFun FDefun {_fInfo, _fArgs, _fBody} _ ->
-      if Set.null failures
-      then for props $
-             verifyFunctionProperty modName funName _fInfo tables _fArgs _fBody
-      else pure [Left (CheckFailure _fInfo (TypecheckFailure failures))]
-    _ -> pure []
```

Comment:
We used to silently ignore failures outside of the `TopFun FDefun{}` case. I am assuming that we probably did not rely on that semantics?

Reply:
This caught my eye as well. I think this is okay but maybe we should discuss.